### ISSUE TITLE
enable more checks for ruff

### ISSUE THREAD

**Issue by prusnak** [lnbits/lnbits#2308]
Currently we enable the F, E, W and I checks for ruff.

There are more we could enable (and fix the reports), such as:
- C - mccabe
- N - naming
- UP - pyupgrade
- B - flake8-bugbear
- A - flake8-builtins
- RUF - ruff specific

More info: https://docs.astral.sh/ruff/rules/

**Comment by dni** [comment]
lets go for it, are you going to try them or should i :)?

**Comment by prusnak** [comment]
> lets go for it, are you going to try them or should i :)?

I already tried some of them.

* `C` is interesting, some things are trivial (like unnecessary `map` usage), some are not that easy (function `pay_invoice` is too complex) which require lot of changes
* `N` is just pain in the ass :D
* `UP` does not respect minimum Python version, so it wants to replace type hints `Union[X, Y]` with `X | Y` which is cool but only supported on Python 3.10 and we still support 3.9
* `B` is mostly about exception handling, might be nice to address the issues
* `A` same as `N`, just pain in the ass
* `RUF` seems nice, good improvements

To recap: I think it is worth fixing issues for `C`, `B` and `RUF`, others not so much.
